internal/app/repositories/file.go

```diff
@@ -95,24 +101,85 @@ func (r *FileRepository) WriteString(record ShorterRecord) error {
 
 }
 
-func (r *FileRepository) Create(fullURL string) (string, error) {
-	var shortURL string
+func (r *FileRepository) GenerateShortURL(ctx context.Context) (string, error) {
+
+	select {
+	case <-ctx.Done():
+		return "", ctx.Err()
+	default:
+	}
 	for {
-		shortURL = randomString()
+		shortURL := randomString()
 		if _, ok := r.store[shortURL]; !ok {
-			r.store[shortURL] = fullURL
-			if err := r.WriteString(ShorterRecord{strconv.Itoa(r.uuid), shortURL, fullURL}); err != nil {
-				return "", err
-			}
 			return shortURL, nil
 		}
 	}
 }
 
-func (r *FileRepository) GetByShort(shortURL string) (string, error) {
+func (r *FileRepository) Create(ctx context.Context, fullURL, shortURL string) (string, error) {
+	select {
+	case <-ctx.Done():
+		return "", ctx.Err()
+	default:
+	}
+	for key, val := range r.store {
+		if val == fullURL {
+			return key, ErrorUniqueViolation
+		}
+	}
+	r.store[shortURL] = fullURL
+	if err := r.WriteString(ShorterRecord{strconv.Itoa(r.uuid), shortURL, fullURL}); err != nil {
+		delete(r.store, shortURL)
+		return shortURL, err
+	}
+	return shortURL, nil
+}
+
+func (r *FileRepository) BatchCreate(ctx context.Context, data []models.BatchCreateRequest) ([]models.BatchCreateResponse, error) {
+	var response []models.BatchCreateResponse
+
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	default:
+	}
+	for _, val := range data {
+		shortID, err := r.GenerateShortURL(ctx)
+		if err != nil {
+			return response, err
+		}
+		shortURL, err := url.JoinPath(r.cfg.BaseURL, shortID)
+		if err != nil {
+			r.logger.Error("ошибка при формировании url", zap.Error(err))
+			return response, err
+		}
+		_, err = r.Create(ctx, val.URL, shortID)
+		if err != nil && !errors.Is(err, ErrorUniqueViolation) {
+			r.logger.Error("ошибка при записи url", zap.Error(err))
+			return response, err
+		}
+		response = append(response, models.BatchCreateResponse{ShortURL: shortURL, UUID: val.UUID})
+	}
+	return response, nil
+}
+
+func (r *FileRepository) GetByShort(ctx context.Context, shortURL string) (string, error) {
+	select {
+	case <-ctx.Done():
+		return "", ctx.Err()
+	default:
+	}
 	fullURL, ok := r.store[shortURL]
 	if ok {
 		return fullURL, nil
 	}
-	return "", errors.New("url not found")
+	return "", NewValueError(shortURL, ErrURLNotFound)
```

Comment:
Я не имел в виду такую обертку, скорее что-то типа:

```suggestion
	return "", fmt.Errorf("%w: %s", ErrURLNotFound, shortURL)
```

Это и добавляет тип к ошибке, и пояснение о ее происхождении